### PR TITLE
Use dynamic require in asc.js without eval

### PR DESCRIPTION
**Pull request by MaxGraey**
Attempt №2

- [x] I've read the contributing guidelines

### CODE REVIEW

File: cli/asc.js [hunk 1]
```diff
@@ -41,28 +41,32 @@ const mkdirp = require("./util/mkdirp");
 const find = require("./util/find");
 const binaryen = global.binaryen || (global.binaryen = require("binaryen"));
 
+const dynRequire = typeof __webpack_require__ === "function"
+  ? __non_webpack_require__
```
[the remark this answers is not in the document]
Comment (dcodeIO) [reply cli/asc.js:45]: So, did you check what's becoming bundled now? Does this not trigger bundling ts-node?

[the remark this answers is not in the document]
Comment (MaxGraey) [reply cli/asc.js:45]: no, I guess better if you check this.

[the remark this answers is not in the document]
Comment (dcodeIO) [reply cli/asc.js:45]: Appears to work, nice. With this in place, we can also get rid of

https://github.com/AssemblyScript/assemblyscript/blob/6df63c46e7bdac143d1babdcff17a73c3c780fa8/webpack.config.js#L70

and

https://github.com/AssemblyScript/assemblyscript/blob/6df63c46e7bdac143d1babdcff17a73c3c780fa8/webpack.config.js#L110

by making sure to always call `dynRequire` for these. Might make sense to refactor this into an `util/node.js` so `util/options.js` can use it as well to get rid of all the webpack warnings.

[the remark this answers is not in the document]
Comment (dcodeIO) [reply cli/asc.js:45]: Btw, you can see what becomes bundled by running `npm run build`. The second list is `asc`.

[the remark this answers is not in the document]
Comment (MaxGraey) [reply cli/asc.js:45]: What about do all this cleanups in separate PR? I worry I could miss something

[the remark this answers is not in the document]
Comment (dcodeIO) [reply cli/asc.js:45]: Made a [PR](https://github.com/MaxGraey/assemblyscript/pull/1) with the suggested changes on your branch :)

**Comment by dcodeIO** [comment]
Thanks :)

**Pull request merged**